src/lr3_comp.erl

```diff
@@ -1,39 +1,27 @@
 %% Copyright (c) 2009, Dave Smith <dizzyd@dizzyd.com> &
 %%                     Tim Dysinger <tim@dysinger.net>
 %% Copyright (c) 2014, 2015 Duncan McGreggor <oubiwann@gmail.com>
+%% Copyright (c) 2016, Eric Bailey <eric@ericb.me>
 %%
 -module(lr3_comp).
 
 -include("lr3_const.hrl").
 
--export([compile/4, compile/5,
+-export([compile/3,
+         compile_dir/4,
          compile_normal_apps/1]).
 
 %% ===================================================================
 %% Public API
 %% ===================================================================
 
-compile(State, Source, AppDir, OutDir) ->
-    rebar_api:debug("\t\tEntered compile/3 ...", []),
-    ErlOpts = rebar_opts:erl_opts(State),
-    compile(State, Source, AppDir, OutDir, ErlOpts).
-
-compile(_State, Source, _AppDir, OutDir, ErlOpts) ->
-    Target = lr3_comp_util:target_file(OutDir, Source),
-    rebar_api:debug("\t\tEntered compile/4 ...", []),
-    rebar_api:debug("\t\tSource: ~p~n\t\tOutDir: ~p", [Source, OutDir]),
-    rebar_api:debug("\t\tErlOpts: ~p", [ErlOpts]),
-    rebar_api:debug("\t\tTarget: ~p", [Target]),
-    %% Make sure that ebin/ exists and is on the path
-    ok = filelib:ensure_dir(filename:join(OutDir, "dummy.beam")),
-    AbsOutDir = filename:absname(OutDir),
-    rebar_api:debug("\t\tAdding ~p to path ...", [AbsOutDir]),
-    true = code:add_patha(AbsOutDir),
-    rebar_api:debug("\t\tCompiling~n\t\t\t~p~n\t\t\tto ~p ...", [Source, Target]),
-    Opts = [{outdir, OutDir}] ++ ErlOpts ++
-       [{i, lr3_comp_util:include_dir()}, return, verbose],
-    rebar_api:debug("\t\tOpts: ~p", [Opts]),
-    CompileResults = lfe_comp:file(Source, Opts),
+compile(Source, Target, Config) ->
+    rebar_api:console(" ~~~~> \tCompiling ~s ...",
+                      [lr3_comp_util:relative(Source)]),
```

Comment:
Wanna fold these online? I mostly introduced the variables to aid in my understanding of the `run/7` semantics.